src/topp/IDFilter.cpp

```diff
@@ -180,29 +182,29 @@ class TOPPIDFilter :
     registerStringList_("blacklist:protein_accessions", "<accessions>", vector<String>(), "All peptides that reference at least one of the provided protein accession are removed.\nOnly proteins not in the provided list are retained.", false);
     registerInputFile_("blacklist:peptides", "<file>", "", "Peptides with the same sequence and modification assignment as any peptide in this file are filtered out. Use with 'blacklist:ignore_modifications' to only compare by sequence.\n", false);
     setValidFormats_("blacklist:peptides", ListUtils::create<String>("idXML"));
-    registerFlag_("blacklist:ignore_modifications", "Compare blacklisted peptides by sequence only.", false);
-    registerStringList_("blacklist:modifications", "<selection>", vector<String>(), "Remove all peptides with sequences that contain (any of) the selected modification(s)", false);
+    registerFlag_("blacklist:ignore_modifications", "Compare blacklisted peptides by sequence only.", true);
+    registerStringList_("blacklist:modifications", "<selection>", vector<String>(), "Remove all peptides with sequences that contain (any of) the selected modification(s)", false, true);
     setValidStrings_("blacklist:modifications", all_mods);
 
     registerTOPPSubsection_("in_silico_digestion", "This filter option removes peptide hits which are not in the list of in silico peptides generated by the rules specified below");
```

Comment:
What is in_silico_digestion filter doing? Isn't it the same as running PeptideIndexer and then filter the unmatched ones?

Reply:
Good question.

Reply:
I think in general there should be no difference, but these two options are not supported by the PeptideIdexer and filtering.

```
  -in_silico_digestion:missed_cleavages <integer>                 Range of allowed missed cleavages in the peptide sequences
                                                                  By default missed cleavages are ignored (default: '-1' min: '-1')
  -in_silico_digestion:methionine_cleavage                        Allow methionine cleavage at the N-terminus of the protein
```